editor/gui/editor_title_bar.cpp

```diff
@@ -80,6 +80,47 @@ void EditorTitleBar::gui_input(const Ref<InputEvent> &p_event) {
 	}
 }
 
+void EditorTitleBar::_notification(int p_what) {
+	if (p_what == NOTIFICATION_SORT_CHILDREN) {
+		Control *prev = nullptr;
+		Control *base = nullptr;
+		Control *next = nullptr;
+		for (int i = 0; i < get_child_count(); i++) {
+			Control *c = as_sortable_control(get_child(i));
+			if (!c) {
+				continue;
+			}
+			if (base) {
+				next = c;
+				break;
+			}
+			if (!c->has_meta("_editor_tab_bar")) {
+				prev = c;
+				continue;
+			}
+			base = c;
+		}
+		if (base && prev && next) {
+			Size2i title_size = get_size();
+			Size2i c_size = base->get_combined_minimum_size();
+
+			int min_offset = prev->get_position().x + prev->get_combined_minimum_size().x;
+			int max_offset = next->get_position().x + next->get_size().x - next->get_combined_minimum_size().x - c_size.x;
+
+			int offset = (title_size.width - c_size.width) / 2;
+			offset = MAX(min_offset, offset);
+			offset = MIN(max_offset, offset);
+
+			Rect2i rect_prev = Rect2i(prev->get_position().x, 0, offset - prev->get_position().x, title_size.height);
+			Rect2i rect_base = Rect2i(offset, 0, c_size.width, title_size.height);
+			Rect2i rect_next = Rect2i(offset + c_size.width, 0, next->get_position().x + next->get_size().x - (offset + c_size.width), title_size.height);
```

Comment:
You could probably assign these directly without variables; each rect is used only once.